met-api/src/met_api/services/widget_service.py

```diff
@@ -32,9 +34,48 @@ def create_widget(widget_data, engagement_id, user_id):
         widget_data['updated_by'] = user_id
         if widget_data.get('engagement_id', None) != int(engagement_id):
             raise ValueError('widget data has engagement id for a different engagement')
-        created_widget = Widget.create_widget(widget_data)
+
+        sort_index = WidgetService._find_higest_sort_index(engagement_id)
+
+        widget_data['sort_index'] = sort_index + 1
+        created_widget = WidgetModel.create_widget(widget_data)
         return WidgetSchema().dump(created_widget)
 
+    @staticmethod
+    def _find_higest_sort_index(engagement_id):
+        # find the highest sort order of the engagement
+        sort_index = 0
+        widgets = WidgetModel.get_widgets_by_engagement_id(engagement_id)
+        if widgets:
+            # Find the largest in the existing widgest
+            sort_index = max(widget.sort_index for widget in widgets)
+        return sort_index
+
+    @staticmethod
+    def sort_widget(engagement_id, widget_items: list, user_id=None):
```

Comment:
I don't think this is still widget_items it's widgets. right?

Reply:
its widget , we are sorting widget

Reply:
so we need to change the paramater name from widget_items to widgets

Reply:
yeah..i misunderstood..now its fixed